### PR TITLE
[fix](nereids)OrToIn and SimplifyRange rules dead loop

### PR DESCRIPTION
**Pull request by englefly**
intro by #41222

[bug]
OrToIn Rule set state flag in Or expression to avoid repeatedly process the same expression. But SimplifyRange Rule may remove this flag, and hence OrToIn and SimplifyRange Rules go into dead loop.

[fix]
if the input expression and output expression of SimplifyRange Rule are the same, SimplifyRange will return input expression to keep expression state.

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/ExpressionOptimization.java [hunk 1]
```diff
@@ -47,14 +47,15 @@ public class ExpressionOptimization extends ExpressionRewrite {
...
-                    OrToIn.INSTANCE,
-                    SimplifyRange.INSTANCE,
                     DateFunctionRewrite.INSTANCE,
                     ArrayContainToArrayOverlap.INSTANCE,
                     CaseWhenToIf.INSTANCE,
                     TopnToMax.INSTANCE,
                     NullSafeEqualToEqual.INSTANCE,
                     LikeToEqualRewrite.INSTANCE
+            ),
+            bottomUp(
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/ExpressionOptimization.java:57]: add comment to explain why this rule need a new bottomUp batch

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run p0

**Comment by englefly** [comment]
run performance

**Comment by englefly** [comment]
run p0

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run feut

**Review by starocean999 — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**